src/main/java/tutorspet/logic/util/AttendanceRecordListUtil.java

```diff
@@ -125,26 +126,50 @@ public static Attendance getAttendanceFromAttendanceRecordList(
     }
 
     /**
-     * Returns an ordered and unmodifiable {@code List<Optional<Attendance>} summarising the {@code targetStudent}'s
-     * attendance in the respective weeks.
+     * Returns the average participation score for {@code targetStudent}.
      */
-    public static List<Optional<Attendance>> getAttendances(AttendanceRecordList targetAttendanceRecordList,
-                                                            Student targetStudent) {
+    public static double getScoreFromAttendance(AttendanceRecordList targetAttendanceRecordList, Student targetStudent)
+            throws CommandException {
         requireAllNonNull(targetAttendanceRecordList, targetStudent);
 
-        List<Optional<Attendance>> studentRecords = new ArrayList<>();
+        List<Optional<Attendance>> listOfAttendance = getAttendances(targetAttendanceRecordList, targetStudent);
 
-        List<AttendanceRecord> attendanceRecords = targetAttendanceRecordList.getAttendanceRecordList();
+        int totalScore = 0;
+        int numOfWeeksParticipated = 0;
 
-        for (AttendanceRecord attendanceRecord : attendanceRecords) {
-            if (attendanceRecord.hasAttendance(targetStudent.getUuid())) {
-                studentRecords.add(Optional.of(attendanceRecord.getAttendance(targetStudent.getUuid())));
-            } else {
-                studentRecords.add(Optional.empty());
+        for (Optional<Attendance> attendance : listOfAttendance) {
+            if (attendance.isPresent()) {
+                totalScore = totalScore + attendance.get().getParticipationScore();
+                numOfWeeksParticipated++;
             }
         }
 
-        return Collections.unmodifiableList(studentRecords);
+        if (numOfWeeksParticipated == 0) {
+            throw new CommandException(MESSAGE_NO_LESSON_ATTENDED);
+        }
+
+        return totalScore / numOfWeeksParticipated;
```

Comment:
I think you missed out on the casting. Might have to tweak some of the test cases also.
```suggestion
        return (double) totalScore / numOfWeeksParticipated;
```